app.js

```diff
@@ -83,16 +83,30 @@ const app = Vue.createApp({
             oi = 0
             for(index in metrics) {
                 [key, value] = metrics[index].split(":")
+
                 expected = Object.entries(this.expectedMetricOrder)[oi++]
-                // There should not be any remaining metrics, or the metric is
-                // not at the right place
-                if(expected == undefined || key != expected[0]) {
-                    console.log("Error invalid vector")
-                    return
+                while(true) {
+                    // If out of possible metrics ordering, it not a valid value thus
+                    // the vector is invalid
+                    if(expected == undefined) {
+                        console.log("Error invalid vector, too many metric values")
+                        return
+                    }
+                    if(key != expected[0]) {
+                        // If not this metric but is mandatory, the vector is invalid
+                        if(oi < 12) {
```

Comment:
Magic number in the code. Should be either constant (in metrics.js?), but more preferable length of some array of mandatory metrics or something similar – however I do not think we have something like this. Also it is unclear why <12 is wrong when based on the comment there are 11 of them?

Reply:
Yep, this is the reason I documented the number of metrics in each group in `metrics.js`. I can change it to `<= 11` if you think it is better for readability, but I don't think a constant would be really interesting if documented properly.

Reply:
The main reason for constant would be to make sure if number of mandatory metric changes (which would never happen I guess) it would be changed also. If there is a comment explaining the number, then I would say it is good enough – even when I personally do not like it.

I am still confused though, if there is 11 mandatory metrics, would not having exactly 11 fine then?

Reply:
In the way CVSS is versionned, changing such a rule would be a breaking change thus a CVSS v5 calculator would have to be built, so it is not subject to changes.
Yes having 11 is valid, this condition is only here to check all the mandatory ones are in the vector and no one is missing before checking for non-mandatory metrics during parsing.

In the latest commit I documented the reason for having `oi <= 11` in the code.

Reply:
OK, I was able to understand it. I think. Basically you say that first 11 metrics have to match the order and exist, otherwise it is a problem.

Do you think it would be possible to formulate it better to make it more clear? It took me quite some time to understand even when reading the comment. Otherwise I can merge it.

Reply:
Sure I can try, but did you check the latest commit (5 hours ago) ? I re-documented it.

Reply:
Now reading it once again it seems clear enough.